crates/iceberg-catalog-bin/src/serve.rs

```diff
@@ -144,9 +144,120 @@ pub(crate) async fn serve(bind_addr: std::net::SocketAddr) -> Result<(), anyhow:
     Ok(())
 }
 
+async fn serve_with_authn<A: Authorizer>(
+    authorizer: A,
+    catalog_state: CatalogState,
+    secrets_state: Secrets,
+    queues: TaskQueues,
+    health_provider: ServiceHealthProvider,
+    listener: tokio::net::TcpListener,
+) -> Result<(), anyhow::Error> {
+    let authn_k8s = if CONFIG.enable_kubernetes_authentication {
+        Some(
+            limes::kubernetes::KubernetesAuthenticator::try_new_with_default_client(
+                Some("kubernetes"),
+                vec![], // All audiences accepted
+            )
+            .await
+            .map_err(|e| {
+                tracing::info!("Failed to create K8s authorizer: {e}");
+                e
+            })
+            .map(|v| {
+                tracing::info!("K8s authorizer created {:?}", v);
+                v
+            })?,
+        )
+    } else {
+        None
+    };
+
+    let authn_oidc = if let Some(uri) = CONFIG.openid_provider_uri.clone() {
+        let mut authenticator = limes::jwks::JWKSWebAuthenticator::new(
+            uri.as_ref(),
+            Some(std::time::Duration::from_secs(3600)),
+        )
+        .await?
+        .set_idp_id("oidc");
```

Comment:
are we using this elsewhere? Should we make it a const / env?

Reply:
Done - its in the bin though, so really only usable there.